src/client/pages/Offer/Introduction/Sidebar/CancellationOptions.tsx

```diff
@@ -76,59 +43,46 @@ const getLabelContent = (textKeys: TextKeyMap, quote: OfferQuote) => {
   })
 }
 
-type QuoteCancellationOptionProps = {
-  quoteCartId: string
-  quote: OfferQuote
-  setShowError: (showError: boolean) => void
+type CancellationOptionsProps = {
+  loadingQuoteIds: Array<string>
+  quotes: BundledQuote[]
+  onToggleCancellationOption: (isChecked: boolean, quoteId: string) => void
 }
 
-const QuoteCancellationOption = ({
-  quoteCartId,
-  quote,
-  setShowError,
-}: QuoteCancellationOptionProps) => {
+export const CancellationOptions = ({
+  loadingQuoteIds,
+  quotes,
+  onToggleCancellationOption,
+}: CancellationOptionsProps) => {
   const textKeys = useTextKeys()
-  const { isoLocale } = useCurrentLocale()
-
-  const [isLoading, setIsLoading] = React.useState(false)
-
-  const [editQuote] = useEditBundledQuoteMutation()
-
-  const isChecked = !quote.startDate
-  const checkboxLabel = getLabelContent(textKeys, quote)
-
-  const handleToggleStartDate = async () => {
-    try {
-      setShowError(false)
-      setIsLoading(true)
-
-      await editQuote({
-        variables: {
-          quoteCartId,
-          locale: isoLocale,
-          quoteId: quote.id,
-          payload: {
-            startDate: isChecked ? formatDate(new Date(), gqlDateFormat) : null,
-          },
-        },
-      })
-    } catch (e) {
-      setShowError(true)
-    } finally {
-      setIsLoading(false)
-    }
-  }
+  const isDisabled = loadingQuoteIds.length > 0
```

Comment:
Now all _cancel my insurance_ checkbox will be disabled when there is one "cancellation" in progress. Not sure if we want that 🤔

Reply:
We do since we use the "createQuoteBundle" mutation. Otherwise you would possibly end up in unintended states.